File: AndroidStudio_Basic _java/Bmi/app/src/main/java/com/ian/bmi/MainActivity.java
package com.ian.bmi;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private EditText editTextWeight;
    private EditText editTextHeight;
        @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
            findView();
        }

    private void findView() {
        editTextWeight = findViewById(R.id.editTextWeight);
        editTextHeight = findViewById(R.id.editTextHeight);
    }

    public void bmi(View view){
    String w =editTextWeight.getText().toString();
    String h =editTextHeight.getText().toString();
    float weight = Float.parseFloat(w);
    float height = Float.parseFloat(h);
    float bmi = weight/(height*height);
    Intent  intent = new Intent(this,RestActivity.class);
    intent.putExtra("BMI",bmi);
    startActivity(intent);
    }
}

File: AndroidStudio_Basic _kotlin/Bmi_Kotlin/app/src/main/java/com/ian/bmi_kotlin/MainActivity.kt
package com.ian.bmi_kotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun bmi(view : View){
        val w = editTextWeight.text.toString().toFloat()
        val h = editTextHeight.text.toString().toFloat()
        print("weight : $w")
        Log.d("MainActivity " ,"weight" + w)
        Log.d("MainActivity " ,"height" + h)
        var  bmi = w/(h*h)
        val intent = Intent(this,ResultActivity::class.java)
        intent.putExtra("Bmi",bmi)
        startActivity(intent)



    }
}
